src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs

```diff
@@ -696,15 +697,25 @@ internal static string GetFileLocationPath(ArtifactLocation artifactLocation, in
                 RunDataCache dataCache = CodeAnalysisResultManager.Instance.RunIndexToRunDataCache[runId];
 
                 Uri uri = artifactLocation.Uri;
-                string uriBaseId = artifactLocation.UriBaseId;
 
+                // try to resolve path using OriginalUriBasePaths
+                string uriBaseId = artifactLocation.UriBaseId;
                 if (!string.IsNullOrEmpty(uriBaseId) && dataCache.OriginalUriBasePaths.ContainsKey(uriBaseId))
                 {
                     Uri baseUri = dataCache.OriginalUriBasePaths[uriBaseId];
                     uri = new Uri(baseUri, uri);
                 }
 
-                path = uri.LocalPath;
+                try
+                {
+                    path = uri.LocalPath;
+                }
+                catch
```

Comment:
>catch [](http://example.com/codeflow?start=16&length=5)

which exception are  we expecting?
can we prevent this checking somehow?

Reply:
If the URI is not a local path e.g. "/example.cs", it throws InvalidOperation when access [LocalPath](https://docs.microsoft.com/en-us/dotnet/api/system.uri.localpath?view=netframework-4.7.2&f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(System.Uri.LocalPath);k(TargetFrameworkMoniker-.NETFramework,Version%253Dv4.7.2);k(DevLang-csharp)%26rd%3Dtrue)

I don't see a method similar to TryGet... or property to check if uri is a local path from Uri.

Reply:
How about adding an extension method bool TryGetLocalPath(out string localPath) to handle this logic?

Reply:
for now, lets not create the extension, since we just have this use-case.
Catch the specific exception and we will be good.

Reply:
updated

---
In reply to: [634683700](https://github.com/microsoft/sarif-visualstudio-extension/pull/398#discussion_r634683700) [](ancestors = 634683700)